client/client.go

```diff
@@ -1583,6 +1586,33 @@ func (c *client) SetExternalTimestamp(ctx context.Context, timestamp uint64) err
 	return nil
 }
 
+func (c *client) GetMinResolvedTS(ctx context.Context, storesID []uint64) (uint64, []*pdpb.StoreMinResolvedTS, error) {
+	if span := opentracing.SpanFromContext(ctx); span != nil {
+		span = opentracing.StartSpan("pdclient.GetMinResolvedTS", opentracing.ChildOf(span.Context()))
+		defer span.Finish()
+	}
+
+	start := time.Now()
+	defer func() { cmdDurationGetMinResolvedTimestamp.Observe(time.Since(start).Seconds()) }()
+
+	protoClient := c.getClient()
+	if protoClient == nil {
+		return 0, nil, errs.ErrClientGetProtoClient
+	}
+	ctx, cancel := context.WithTimeout(ctx, c.option.timeout)
+	defer cancel()
+	ctx = grpcutil.BuildForwardContext(ctx, c.GetLeaderAddr())
```

Comment:
What about putting this before `WithTimeout`?

Reply:
Maybe a new pr is needed to change all the similar formatting 🤔
I will change here.